router next documentation (#6045)
apollographql/router: docs/source/configuration/overview.mdx

- You can configure a Redis-backed _distributed_ cache that enables multiple router instances to share cached values. For details, see [Distributed caching in the GraphOS Router](./distributed-caching/).
- You can configure a Redis-backed _entity_ cache that enables a client query to retrieve cached entity data split between subgraph reponses. For details, see [Subgraph entity caching in the GraphOS Router](./entity-caching/).

<MinVersion version ="1.49.0">

### Native query planner

</MinVersion>

Starting with v1.49.0, the router can run a Rust-native query planner. This native query planner can be run by itself to plan all queries, replacing the legacy Javascript implementation.

<Tip>

> Starting with v2.0.0, to run the most performant and resource-efficient native query planner, set the following options in your `router.yaml`:

Reviewer: So the configuration below not only enables native query planning, but also does not start up the v8, the javascript runtime. We should note that in this paragraph. But do we need to mention the javascript runtime a bit earlier in this section? will it be too confusing it's brought up here? or should the native query planner section just take care of that?
```suggestion
Starting with v1.56.0, to run the most performant and resource-efficient native query planner and to disable the JavaScript runtime in the router, set the following options in your `router.yaml`:
```

Reviewer 2: Will describe the JS runtime earlier

```yaml title="router.yaml"

> supergraph:
>   generate_query_fragments: true
> 

Reviewer: let's drop this for this configuration, but mention it as an additional configuration that makes native query planning more optimal?
```suggestion
```

- You can configure a Redis-backed _distributed_ cache that enables multiple router instances to share cached values. For details, see [Distributed caching in the GraphOS Router](./distributed-caching/).
- You can configure a Redis-backed _entity_ cache that enables a client query to retrieve cached entity data split between subgraph reponses. For details, see [Subgraph entity caching in the GraphOS Router](./entity-caching/).

<MinVersion version ="1.49.0">

### Native query planner

</MinVersion>

> Starting with v1.49.0, the router can run a Rust-native query planner. This native query planner can be run by itself to plan all queries, replacing the legacy Javascript implementation.

Reviewer: ```suggestion
Starting with v1.49.0, the router can run a Rust-native query planner. This native query planner can be run by itself to plan all queries, replacing the legacy JavaScript implementation.
```
